File: pset6/credit/credit.py
from cs50 import get_int
num = get_int("Number: ")

n = num
even = 0
odd = 0
while True:
    even += n % 10
    n = int(n / 10)

    if n == 0:
        break
    if len(str(n)) == 2:
        twodig = n
    if len(str(n)) == 1:
        onedig = n

    m = (n %  10)*2
    odd += m % 10
    odd += int(m / 10)
    n = int(n / 10)

    if n == 0:
        break
    if len(str(n)) == 2:
        twodig = n
    if len(str(n)) == 1:
        onedig = n

if ((even + odd) % 10) == 0:
    if (len(str(num)) == 15 and twodig in [34, 37]):
        print("AMEX")
    elif (len(str(num)) == 16 and twodig in [51, 52, 53, 54, 55]):
        print("MASTERCARD")
    elif (len(str(num)) in [13, 16]  and onedig == 4):
        print("VISA")
    else:
        print("INVALID")
else:
    print("INVALID")

File: pset1/credit/credit.c
#include <stdio.h>
#include <cs50.h>

int sum(long);
int dg(long);
int td(long);
int od(long);


int main()
{
    long x = get_long("Number: ");
    int s = sum(x);
    int dig = dg(x);
    int twodig = td(x);
    int onedig = od(x);

    if (s % 10 == 0)
    {
        //AMEX
        if (dig == 15 && (twodig == 34 || twodig == 37))
        {
            printf("AMEX\n");
        }

        //MASTERCARD
        else if (dig == 16 && (twodig == 51 || twodig == 52 || twodig == 53 || twodig == 54 || twodig == 55))
        {
            printf("MASTERCARD\n");
        }

        //VISA
        else if ((dig == 13 || dig == 16) && onedig == 4)
        {
            printf("VISA\n");
        }

        else
        {
        printf("INVALID\n");

        }
    }

    else
    {
        printf("INVALID\n");
    }

}

//to find sum
int sum(long n)
{
    int sum_odd = 0;
    int sum_even = 0;

    for (int i = 1; n > 0 ; i++)
    {
       //odd sum
       sum_odd+= n % 10;

       //to get to next digit
       n-= n % 10;
       n /= 10;

       //even sum
       int m = (n % 10)*2;
       sum_even += m % 10;
       m-= m % 10;
       m /= 10;
       sum_even += m %10;

        //to get to next digit
        n-= n % 10;
        n/= 10;
    }

    return sum_odd + sum_even;
}


// to find number of digits in the number
int dg(long n)
{
    int dig;
    for(dig = 0; n > 0; dig++)
    {
        n-= n % 10;
        n/= 10;
    }

    return dig;
}


//to find first two digits
int td(long n)
{
    for (int i = 1; n > 99; i++)
    {
        n-= n % 10;
        n/= 10;
    }
    return n;
}


//to find first one digit
int od(long n)
{
    for (int i = 1; n > 9; i++)
    {
        n-= n % 10;
        n/= 10;
    }
    return n;
}
